pkg/webhook/policy.go

```diff
@@ -148,8 +146,12 @@ func (h *validationHandler) Handle(ctx context.Context, req admission.Request) a
 	requestResponse := unknownResponse
 	defer func() {
 		if h.reporter != nil {
-			if err := h.reporter.ReportAdmissionRequest(
-				requestResponse, time.Since(timeStart)); err != nil {
+			if err := h.reporter.ReportRequest(
+				requestResponse, admissionStatusKey, responseTimeInSecM.M(time.Since(timeStart).Seconds())); err != nil {
+				log.Error(err, "failed to report request")
+			}
+			if err := h.reporter.ReportRequest(
```

Comment:
We should collapse these into two functions:

`ReportValidationRequest`

and 

`ReportMutationRequest`

That way we can do the fanout into two metrics internally in the metrics interface. That will make a future removal of the deprecated metric easier.

Reply:
This PR already had separate functions for this. Please look [here](https://github.com/open-policy-agent/gatekeeper/pull/976#discussion_r533124875)

Reply:
ack, moving this discussion there.